File: Data Structures & Algorithms/three-integer-sum/submission-6.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ans=[]
        nums.sort()

        for i,ele in enumerate(nums):
            if i>0 and nums[i]==nums[i-1]:
                continue
            if nums[i]>0:
                break
            
            j=i+1
            k=len(nums)-1
            while(j<k):
                sum = nums[i] + nums[j] + nums[k]
                if sum==0:
                    ans.append([nums[i], nums[j], nums[k]])
                    j+=1
                    k-=1
                    while j<k and nums[j] == nums[j-1]:
                        j+=1
                elif sum>0:
                    k-=1
                else:
                    j+=1
        return ans
            

        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int ele: nums){
            set.add(ele);
        }

        int ans=0;

        for(int ele: nums){
            if(!set.contains(ele-1)){
                int start=ele;
                int count=0;
                while(set.contains(start)){
                    start++;
                    count++;
                }
                ans=Math.max(ans, count);
            }
        }
        return ans;
    }
}
